sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options for logging within a <see cref="ClientPipeline"/>.
+/// </summary>
+public class ClientLoggingOptions
+{
+    private bool _frozen;
+
+    private bool? _enableLogging;
+    private bool? _enableMessageLogging;
+    private bool? _enableMessageContentLogging;
+    private int? _messageContentSizeLimit;
+    private ILoggerFactory? _loggerFactory;
+    private PipelineMessageSanitizer? _sanitizer;
+    private ChangeTrackingStringList? _allowedHeaderNames;
+    private ChangeTrackingStringList? _allowedQueryParameters;
+
+    private static HashSet<string> DefaultAllowedHeaderNames { get; } = [
+            "traceparent",
+            "Accept",
+            "Cache-Control",
+            "Connection",
+            "Content-Length",
+            "Content-Type",
+            "Date",
+            "ETag",
+            "Expires",
+            "If-Match",
+            "If-Modified-Since",
+            "If-None-Match",
+            "If-Unmodified-Since",
+            "Last-Modified",
+            "Pragma",
+            "Request-Id",
+            "Retry-After",
+            "Server",
+            "Transfer-Encoding",
+            "User-Agent",
+            "WWW-Authenticate" ];
+    private static HashSet<string> DefaultAllowedQueryParameters { get; } = ["api-version"];
+    private static PipelineMessageSanitizer DefaultSanitizer { get; } = new(DefaultAllowedQueryParameters, DefaultAllowedHeaderNames);
+
+    internal const bool DefaultEnableLogging = true;
+    internal const bool DefaultEnableMessageContentLogging = false;
+    internal const int DefaultMessageContentSizeLimit = 4 * 1024;
```

Comment:
Should we include "bytes" or whatever unit in the name to avoid confusion?

Reply:
Thoughts on whether I should I add this to the property name too? I think it's fine as is, for ref in core it is [`LoggedContentSizeLimit `](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L103), but I wonder if it might be slightly more clear?
`ClientLoggingOptions.MessageContentSizeLimit` => `ClientLoggingOptions.MessageContentSizeLimitBytes`

Reply:
I would probably do both, but I don't feel strongly about it.